### PR TITLE
ENG-16199 TTL migrate: only insert to shadow stream on partition 0 from replicated source table

### PR DESCRIPTION
**Pull request by zeemanhe**
Updates migrating indexes after constraint violation is checked. Also used MP partition id as a partition key for migrated row deletion.

### CODE REVIEW

**Comment by zeemanhe** [comment]
[Test case update](https://github.com/VoltDB/pro/pull/2621)

File: src/ee/storage/persistenttable.cpp [hunk 1]
```diff
@@ -889,6 +880,14 @@ void PersistentTable::doInsertTupleCommon(TableTuple& source, TableTuple& target
         target.setDirtyFalse();
     }
 
+    // add it to migrating index
```
Comment (luyangco) [inline src/ee/storage/persistenttable.cpp]: Why move this block down to here?

Comment (zeemanhe) [reply src/ee/storage/persistenttable.cpp]: move to after conflict resolution is checked.

File: src/ee/storage/persistenttable.cpp [hunk 2]
```diff
@@ -1140,8 +1143,10 @@ void PersistentTable::updateTupleWithSpecificIndexes(TableTuple& targetTupleToUp
     if (fromMigrate) {
         assert(isTableWithMigrate(m_tableType) && m_shadowStream != nullptr);
         migratingAdd(ec->currentSpHandle(), targetTupleToUpdate);
-        m_shadowStream->streamTuple(sourceTupleWithNewValues, ExportTupleStream::MIGRATE,
-                doDRActions(drStream) ? drStream : NULL);
+        // add to shadow stream if the table is partitioned or partition 0 for replicated table
+        if (!isReplicatedTable() || ec->getPartitionId() == 0) {
```
Comment (wweiss-voltdb) [inline src/ee/storage/persistenttable.cpp]: This code also is needed in deleteTuple

File: src/ee/storage/persistenttable.cpp (hunk 1 above)
Comment (wweiss-voltdb) [inline src/ee/storage/persistenttable.cpp]: Change comment: 
// add it to migrating index when loading tuple from a recover or rejoin snapshot (only)

File: src/ee/storage/persistenttable.cpp (hunk 2 above)
Comment (wweiss-voltdb) [inline src/ee/storage/persistenttable.cpp]: also wrap the following code above (at 1024):
           // We assume that only fallible and undoable UPDATEs should be propagated to the EXPORT Shadow Stream
           if (isTableWithExportUpdateOld(m_tableType)) {
               m_shadowStream->streamTuple(targetTupleToUpdate, ExportTupleStream::STREAM_ROW_TYPE::UPDATE_OLD);
           }
           if (isTableWithExportUpdateNew(m_tableType)) {
               m_shadowStream->streamTuple(sourceTupleWithNewValues, ExportTupleStream::STREAM_ROW_TYPE::UPDATE_NEW);
           }

with:  if (!isReplicatedTable() || ec->getPartitionId() == 0) {

Comment (zeemanhe) [reply src/ee/storage/persistenttable.cpp]: added

Comment (zeemanhe) [reply src/ee/storage/persistenttable.cpp]: updated.

**Pull request merged**